runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillStateReader.java

```diff
@@ -577,9 +655,30 @@ public void addWeighted(T elem, long weight) {
     return valueList;
   }
 
-  private <T> void consumeBag(TagBag bag, StateTag stateTag) {
+  private <T> List<TimestampedValue<T>> sortedListPageValues(
+      Windmill.TagSortedListFetchResponse sortedListFetchResponse, Coder<T> elemCoder) {
+    if (sortedListFetchResponse.getEntriesCount() == 0) {
+      return new WeightedList<>(Collections.emptyList());
+    }
+
+    WeightedList<TimestampedValue<T>> entryList =
+        new WeightedList<>(new ArrayList<>(sortedListFetchResponse.getEntriesCount()));
+    for (SortedListEntry entry : sortedListFetchResponse.getEntriesList()) {
+      try {
+        T value = elemCoder.decode(entry.getValue().newInput(), Coder.Context.OUTER);
+        entryList.addWeighted(
+            TimestampedValue.of(value, Instant.ofEpochMilli(entry.getSortKey() / 1000)),
```

Comment:
Use WindmillTimeUtils.windmillToHarnessTimestamp().  Current code is wrong for negative timestamps.

Reply:
done

Reply:
done